compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/descriptors/JvmDeclarationFactory.kt

```diff
@@ -133,6 +135,23 @@ class JvmDeclarationFactory(
             }
         }
 
+    fun getPrivateFieldForObjectInstance(singleton: IrClass): IrField =
+        if (singleton.isCompanion && singleton.parentAsClass.isJvmInterface)
+            interfaceCompanionFieldDeclarations.getOrPut(singleton) {
+                buildField {
+                    name = Name.identifier("\$\$INSTANCE")
+                    type = singleton.defaultType
+                    origin = JvmLoweredDeclarationOrigin.INTERFACE_COMPANION_INSTANCE
+                    isFinal = true
+                    isStatic = true
+                    visibility = JavaVisibilities.PROTECTED_AND_PACKAGE
```

Comment:
Please check, but seems old backend uses PACKAGE visibility

Reply:
Yes, there should be no access flag. For some reason I thought PROTECTED_AND_PACKAGE translates to that, thanks.